extensions/amp-analytics/0.1/test/test-resource-timing.js

```diff
@@ -0,0 +1,334 @@
+    responseStart: startTime + dnsTime + tcpTime + serverTime,
+    responseEnd: startTime + dnsTime + tcpTime + serverTime + transferTime,
+    decodedBodySize: bodySize,
+    encodedBodySize: bodySize * 0.7,
+    transferSize: cached ? 0 : bodySize * 0.7 + 200, // +200 for header size
+  };
+};
+
+describes.fakeWin('resourceTiming', {amp: true}, env => {
+  let win;
+
+  /**
+   * @param {!Array<!PerformanceResourceTiming} fakeEntries
+   * @param {!JsonObject} resourceTimingSpec
+   * @param {string} expectedResult
+   * @return {!Promise<undefined>}
+   */
+  const runSerializeTest = function(
+    fakeEntries, resourceTimingSpec, expectedResult) {
+    sandbox.stub(win.performance, 'getEntriesByType').returns(fakeEntries);
+    return serializeResourceTiming(resourceTimingSpec, win).then(result => {
+      expect(result).to.equal(expectedResult);
+    });
+  };
+
+  beforeEach(() => {
+    win = env.win;
+    win.performance = window.performance;
+    installVariableService(win);
+  });
+
+  it('should return empty if the performance API is not supported', () => {
+    const fakeWin = {};
+    return serializeResourceTiming(newResourceTimingSpec(), fakeWin)
+        .then(result => {
+          expect(result).to.equal('');
+        });
+  });
+
+  it('should return empty when resource timing is not supported', () => {
+    // Performance API (fakeWin.performance)  oesn't support resource timing.
+    const fakeWin = {performance: {}};
+    return serializeResourceTiming(newResourceTimingSpec(), fakeWin)
+        .then(result => {
+          expect(result).to.equal('');
+        });
+  });
+
+  it('should return empty if resourceTimingSpec is empty', () => {
+    const entry = newPerformanceResourceTiming(
+        'http://foo.example.com/lib.js?v=123', 'script', 100, 500, 10 * 1000,
+        false);
+    return runSerializeTest([entry], {}, '');
+  });
+
+  it('should return empty if encoding spec is empty', () => {
+    const entry = newPerformanceResourceTiming(
+        'http://foo.example.com/lib.js?v=123', 'script', 100, 500, 10 * 1000,
+        false);
+    const spec = newResourceTimingSpec();
+    delete spec['encoding'];
+    return runSerializeTest([entry], spec, '');
+  });
+
+  it('should return empty if encoding spec is missing delim', () => {
+    const entry = newPerformanceResourceTiming(
+        'http://foo.example.com/lib.js?v=123', 'script', 100, 500, 10 * 1000,
+        false);
+    const spec = newResourceTimingSpec();
+    delete spec['encoding']['delim'];
+    return runSerializeTest([entry], spec, '');
+  });
+
+  it('should return empty if encoding spec is missing entry', () => {
+    const entry = newPerformanceResourceTiming(
+        'http://foo.example.com/lib.js?v=123', 'script', 100, 500, 10 * 1000,
+        false);
+    const spec = newResourceTimingSpec();
+    delete spec['encoding']['entry'];
+    return runSerializeTest([entry], spec, '');
+  });
+
+  it('should serialize matching entries', () => {
```

Comment:
Could you please add a test that the resourceTiming value has been encoded properly. For example with `delim=?`

Reply:
Done